samples/cellular/modem_shell/src/utils/mosh_print.c

```diff
@@ -197,4 +200,30 @@ void mosh_print_version_info(void)
 #else
 	printk("\nMOSH build variant: dev\n\n");
 #endif
+
+#if defined(CONFIG_MODEM_INFO)
```

Comment:
Do we need this flagging? There are modem_info usages without `CONFIG_MODEM_INFO` flagging so you cannot compile MOSH without it anyway. So either flagging should be removed or made complete.

Reply:
I guess the original intention has been, that mosh could be compiled without `link` support, and `MOSH_LINK` depends on `MODEM_INFO`. But I doubt that link functionality can really be disabled, and not sure if it really makes much sense considering the nature of the sample. I tried compiling with `CONFIG_MOSH_LINK=n` and there are at least several failures when trying to include `link_api.h`.

I can remove the flagging from the PR, but should I also do some cleanup for this while at it?